lib/client.js

```diff
@@ -192,6 +254,172 @@ Client.prototype = {
     this.postMessage(requestKey, options);
 
     return deferred.promise;
+  },
+
+  /// #### client.metadata()
+  ///
+  /// Request metadata for the app, such as the app ID and installation ID.
+  ///
+  /// ##### Returns
+  ///
+  /// A [Promises/A+](https://promisesaplus.com) conformant `promise` object.
+  ///
+  /// ```javascript
+  /// var client = ZAFClient.init();
+  ///
+  /// client.metadata().then(function(metadata) {
+  ///   console.log(metadata); // { appId: 1234, installationId: 1234 }
+  /// });
+  /// ```
+  metadata: function() {
+    var deferred = Promise.defer();
+    if (this._metadata) {
+      deferred.resolve(this._metadata);
+    } else {
+      this.on('app.registered', function() {
+        deferred.resolve(this._metadata);
+      }.bind(this));
+    }
+    return deferred.promise;
+  },
+
+  /// #### client.context()
+  ///
+  /// Request context for the app, such as the host and location.
+  /// Depending on the location, the context will provide additional identifiers
+  /// which can be used with the REST API to request additional data.
+  ///
+  /// ##### Returns
+  ///
+  /// A [Promises/A+](https://promisesaplus.com) conformant `promise` object.
+  ///
+  /// ```javascript
+  /// var client = ZAFClient.init();
+  ///
+  /// client.metadata().then(function(metadata) {
+  ///   console.log(metadata); // { host: 'zendesk', hostAccountId: 'mysubdomain', location: 'ticket_sidebar', ticketId: 1234 }
+  /// });
+  /// ```
+  context: function() {
+    var deferred = Promise.defer();
+    if (this._context) {
+      deferred.resolve(this._context);
+    } else {
+      this.on('app.registered', function() {
+        deferred.resolve(this._context);
+      }.bind(this));
+    }
+    return deferred.promise;
+  },
+
+  /// #### client.get(paths)
+  ///
+  /// Get data from the UI asynchronously.
+  ///
+  /// ##### Arguments
+  ///
+  ///   * `paths` an array of path strings or a single path string
+  ///
+  /// ##### Returns
+  ///
+  /// A [Promises/A+](https://promisesaplus.com) conformant `promise` object.
+  ///
+  /// ```javascript
+  /// var client = ZAFClient.init();
+  ///
+  /// client.ui(['ticket.subject', 'ticket.fields.custom_field_1234.value').then(function(data) {
```

Comment:
Doc should be `get`, not `ui`.  Also, missing a closing square bracket after the `'ticket.fields.custom_field_1234.value'`